spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/RabbitProperties.java

```diff
@@ -282,6 +289,28 @@ public void setVirtualHost(String virtualHost) {
 		this.virtualHost = "".equals(virtualHost) ? "/" : virtualHost;
 	}
 
+	public String getAddressShuffleMode() {
+		return this.addressShuffleMode;
+	}
+
+	/**
+	 * If addresses have been set and address shuffle mode has been set it is returned.
+	 * Otherwise returns the result of calling {@code getAddressShuffleMode()}.
+	 * @return the address shuffle mode
+	 * @see #setAddressShuffleMode(String)
+	 * @see #getAddressShuffleMode()
+	 */
+	public AbstractConnectionFactory.AddressShuffleMode determineAddressShuffleMode() {
```

Comment:
I don't think we should do that. Rather we should set whatever has been set by the user.